src/lib/y2storage/proposal/devices_planner.rb

```diff
@@ -215,22 +215,17 @@ def home_device
         home_vol
       end
 
-      def remove_shadowed_subvols!(planned_devices)
+      def remove_shadowed_subvols(planned_devices)
         planned_devices.each do |device|
-          next unless device.respond_to?(:remove_shadowed_subvolumes!)
+          next unless device.respond_to?(:subvolumes)
 
-          other_devices = planned_devices.reject { |dev| dev == device }
-          mount_points = other_devices.map { |dev| mount_point_for(dev) }.compact
-          device.remove_shadowed_subvolumes!(mount_points)
+          device.shadowed_subvolumes(planned_devices).each do |subvol|
+            log.info "Subvolume #{subvol} would be shadowed. Removing it."
+            device.subvolumes.delete(subvol)
+          end
```

Comment:
Yes, in the roadmap there is room for a lot of unification there. Starting with the devices creators and continuing with these planners.